presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -134,6 +137,12 @@ private RubixConfigurationInitializer initializeRubix(RubixConfig rubixConfig)
 
         // initialize rubix in master-only mode
         rubixConfig.setStartServerOnCoordinator(true);
+
+        // use different rubix ports for each test to prevent nondeterministic bind errors
+        rubixConfig.setBookKeeperServerPort(DEFAULT_BOOKKEEPER_SERVER_PORT + cachingFileSystemCounter);
+        rubixConfig.setDataTransferServerPort(DEFAULT_DATA_TRANSFER_SERVER_PORT + cachingFileSystemCounter);
```

Comment:
The test class is executed single threaded, so previously allocated ports should be free to take. 
If this is not the case, this means that resource cleanup is ineffective, or async. It shouldn't be.

Reply:
> The test class is executed single threaded, so previously allocated ports should be free to take.

Tests is single-threaded, but Rubix servers are not. Added waiting for Rubix server to stop on shutdown